ethereum/stratum/src/test/java/org/hyperledger/besu/ethereum/stratum/StratumServerTest.java

```diff
@@ -47,16 +72,194 @@ public void runJobSetZeroDifficultyShouldNotThrow() {
+                  NetClient client = vertx.createNetClient();
+                  client
+                      .connect(server.actualPort(), host)
+                      .onComplete(
+                          testContext.succeeding(
+                              socket -> {
+                                socket.handler(
+                                    resp -> {
+                                      testContext.verify(
+                                          () -> {
+                                            assertThat(resp.length()).isNotZero();
+                                            assertThat(
+                                                    resp.toString(StandardCharsets.UTF_8)
+                                                        .split(System.lineSeparator()))
+                                                .hasSize(2);
+                                            stratum
+                                                .stop()
+                                                .onComplete(testContext.succeedingThenComplete());
+                                          });
+                                    });
+                                socket
+                                    .write(
+                                        DEFAULT_TEST_MESSAGE
+                                            .copy()
+                                            .appendString(System.lineSeparator())
+                                            .appendBuffer(DEFAULT_TEST_MESSAGE)
+                                            .appendString(System.lineSeparator()))
+                                    .onComplete(
+                                        writeResult ->
+                                            testContext.verify(
+                                                () -> assertThat(writeResult.failed()).isFalse()));
+                              }));
+                }));
+  }
+
+  @Test
+  public void handleHttpPost(final Vertx vertx, final VertxTestContext testContext) {
+    String host = "localhost";
+    StratumServer stratum = new StratumServer(vertx, mockPoW, 0, host, "", new NoOpMetricsSystem());
+    stratum
+        .start()
+        .onComplete(
+            testContext.succeeding(
+                server -> {
+                  WebClient http =
+                      WebClient.create(
+                          vertx,
+                          new WebClientOptions()
+                              .setDefaultHost(host)
+                              .setDefaultPort(server.actualPort()));
+                  http.post("/")
+                      .sendBuffer(DEFAULT_TEST_MESSAGE)
+                      .onComplete(
+                          testContext.succeeding(
+                              resp -> {
+                                testContext.verify(
+                                    () -> {
+                                      assertThat(resp.statusCode()).isEqualTo(200);
```

Comment:
Can we assert on the response body here?

Reply:
I added the assert on the response body expecting the same value used in `StratumConnectionTest#testHttpMessage`